sdk/location/src/main/java/com/klaviyo/location/KlaviyoLocationManager.kt

```diff
@@ -7,27 +7,55 @@ import android.content.Context
 import android.content.Intent
 import androidx.annotation.RequiresPermission
 import com.google.android.gms.location.Geofence
+import com.google.android.gms.location.Geofence.NEVER_EXPIRE
 import com.google.android.gms.location.GeofenceStatusCodes
 import com.google.android.gms.location.GeofencingClient
 import com.google.android.gms.location.GeofencingEvent
 import com.google.android.gms.location.GeofencingRequest
 import com.google.android.gms.location.LocationServices
+import com.klaviyo.analytics.networking.ApiClient
+import com.klaviyo.analytics.networking.requests.FetchGeofencesResult
 import com.klaviyo.core.Registry
+import java.util.Collections
+import java.util.concurrent.CopyOnWriteArrayList
+import org.json.JSONArray
+
+/**
+ * Extension property to access a shared [LocationManager] instance
+ */
+internal val Registry.locationManager: LocationManager
```

Comment:
Because a geofence event intent could cold-start the app, we can't rely on the application code having already run `Klaviyo.registerGeofences` so we need a fallback for creating this service.